src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -40,6 +42,7 @@ internal enum PreconditionState
             ActionContext context,
             FileResult result,
             long? fileLength,
+            bool isRangeRequest = true,
```

Comment:
enableRangeProcessing = true
Why not add it at the end? Then you don't have to change PhysicalFile or VirtualFile.